packages/suite-web/e2e/tests/metadata/wallet-metadata.test.ts

```diff
@@ -2,153 +2,154 @@
-            cy.getTestElement(
-                `@metadata/walletLabel/${secondHiddenWalletState}/add-label-button`,
-            ).click();
-            cy.getConfirmActionOnDeviceModal();
-            cy.task('pressYes'); // only now labeling was enabled
-            cy.getTestElement('@metadata/input').type(
-                'still works, metadata enabled for currently not selected device{enter}',
-            );
-
-            cy.window()
-                .its('store')
-                .invoke('getState')
-                .then(state => {
-                    console.log(state);
-                    const errors = state.notifications.filter(
-                        (n: { type: string }) => n.type === 'error',
+
+            it('labels can be enabled and edited when different wallet is open', () => {
+                cy.step('Setup standard wallet with label and edit it', () => {
+                    onAccountsPage.openBtcAccount(1);
+                    onMenu.openSwitchDevice();
+
+                    onSwitchDeviceModal.clickAddLabel(standardWalletIndex);
+                    cy.passThroughInitMetadata(MetadataProvider.Dropbox);
+                    onSwitchDeviceModal.typeLabel('label for standard wallet');
+                });
+
+                cy.step('Add passphrase wallet C and switch back to first wallet', () => {
+                    cy.addHiddenWallet('C');
+                    onMenu.openSwitchDevice();
+                    handleLabelingOnDevice('pressNo');
+                    onSwitchDeviceModal.openDevice(standardWalletIndex);
+                });
+
+                cy.step('Enable labeling for wallet C', () => {
+                    onMenu.openSwitchDevice();
+                    onSwitchDeviceModal.clickAddLabel(hiddenWalletIndex);
+
+                    handleLabelingOnDevice('pressYes');
+                    onSwitchDeviceModal.typeLabel(
+                        'still works, metadata enabled for currently not selected device',
                     );
+                });
 
-                    return expect(errors).to.be.empty;
+                cy.step('Verify wallet label', () => {
+                    onSwitchDeviceModal
+                        .getLabel(hiddenWalletIndex)
+                        .should(
+                            'contain',
+                            'still works, metadata enabled for currently not selected device',
+                        );
                 });
+
+                checkStateNotificationsForErrors();
+            });
         });
     });
 });
 
+function openApp(): void {
+    cy.prefixedVisit('/', {
+        onBeforeLoad: (win: Window) => {
+            cy.stub(win, 'open').callsFake(stubOpen(win));
+            cy.stub(win, 'fetch').callsFake(rerouteMetadataToMockProvider);
+        },
+    });
+}
+
+function handleLabelingOnDevice(action: 'pressYes' | 'pressNo'): void {
+    cy.getConfirmActionOnDeviceModal();
+    cy.wait(501);
+    cy.task(action);
+}
```

Comment:
this doesn't feel like useful abstraction to me. at least `handleLabelingOnDevice` should be renamed to something like `waitForDeviceInteractionAndConfirm` or similar. It is not exclusive to labeling.

Reply:
aha, I see, it is defined solely in wallet-metadata. file. which makes it ok.